src/masternode/payments.cpp

```diff
@@ -24,24 +24,32 @@
 #include <cassert>
 #include <string>
 
-[[nodiscard]] static bool GetBlockTxOuts(const CBlockIndex* const pindexPrev, const CAmount blockSubsidy, const CAmount feeReward, std::vector<CTxOut>& voutMasternodePaymentsRet)
+CAmount PlatformShare(const CAmount reward)
```

Comment:
Only most of the header contents of `masternode/payments.h` are moved to `evo/cshelper.h`, the contents of `masternode/payments.cpp` and `evo/specialtxman.cpp` remain where they are and are only modified to the extent that the new name was being used.

The new header exists somewhat document that `ChainstateHelper` is just an vessel for logic implemented in those two files and the logic implemented in `specialtxman.cpp` should not be considered a part of masternode payment code (which would've been in the implication had the special transaction functions been defined in `masternode/payments.h`).

I kept it at the top because it's a relatively small function (and now additionally, because it's the only function actually defined in its corresponding header, the rest being implementations of the definitions now in `cshelper.h`)